kotsadm/pkg/reporting/reporting.go

```diff
@@ -128,3 +134,83 @@ func getK8sVersion() (string, error) {
 	}
 	return k8sVersion.GitVersion, nil
 }
+
+func SendPreflightsReportToReplicatedApp(license *kotsv1beta1.License, appSlug string, clusterID string, sequence int, skipPreflights bool, installStatus string) error {
+	urlValues := url.Values{}
+
+	sequenceToStr := fmt.Sprintf("%d", sequence)
+	skipPreflightsToStr := fmt.Sprintf("%t", skipPreflights)
+
+	urlValues.Set("sequence", sequenceToStr)
+	urlValues.Set("skipPreflights", skipPreflightsToStr)
+	urlValues.Set("installStatus", installStatus)
+
+	url := fmt.Sprintf("%s/preflights/reporting/%s/%s?%s", license.Spec.Endpoint, appSlug, clusterID, urlValues.Encode())
+
+	var buf bytes.Buffer
+	postReq, err := http.NewRequest("POST", url, &buf)
+	if err != nil {
+		return errors.Wrap(err, "failed to call newrequest")
+	}
+	postReq.Header.Add("Authorization", license.Spec.LicenseID)
+	postReq.Header.Set("Content-Type", "application/json")
+
+	resp, err := http.DefaultClient.Do(postReq)
+	if err != nil {
+		return errors.Wrap(err, "failed to send preflights reports")
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != 201 {
+		return errors.Errorf("Unexpected status code %d", resp.StatusCode)
+	}
+
+	_, err = ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return errors.Wrap(err, "failed to read")
+	}
+
+	return nil
+}
+
+func PreflightInfoThreadSend(appID string, appSlug string, sequence int, isSkipPreflights bool, isFailedPreflight bool) error {
```

Comment:
this function never returns (it will keep running forever in some cases). i suggest checking the application deployment status in the go routine (thread), and only report when the application deployment process is done and has a status like "failed, "deployed", etc...)